docs/intro/tutorial.rst

```diff
@@ -1,141 +1,434 @@
+        class.
+
+    -   Provides attributes like :attr:`~web_poet.pages.WebPage.html` and
+        :attr:`~web_poet.pages.WebPage.url`, and methods like
+        :meth:`~web_poet.pages.WebPage.css`,
+        :meth:`~web_poet.pages.WebPage.urljoin`, and
+        :meth:`~web_poet.pages.WebPage.xpath`, that make it easier to write
+        web data extraction code.
+
+-   ``BookPage`` declares ``Book`` as its return type.
+
+    :class:`~web_poet.pages.WebPage`, like its parent class
+    :class:`~web_poet.pages.ItemPage`, is a :ref:`generic class <generics>`
+    that accepts a type parameter. Unlike most generic classes, however, the
+    specified type parameter is used for more than type hinting: it determines
+    the item class that is used to store the data that fields return.
+
+-   ``BookPage`` is decorated with :meth:`~web_poet.overrides.handle_urls`,
+    which indicates for which URLs ``BookPage`` is intended to work.
+
+-   ``BookPage`` defines a field named ``title``.
+
+    :ref:`Fields <fields>` are methods of page object classes, preferably async
+    methods, decorated with the :meth:`~web_poet.fields.field` decorator.
+    Fields define the logic to extract a specific piece of information from the
+    input of your page object class.
+
+    ``BookPage.title`` extracts the title of a book from a book details
+    webpage. Specifically, it extracts the text from the first ``h1`` element
+    on the input HTTP response.
+
+    Here ``title`` is not an arbitrary name, it was chosen specifically to
+    match ``Book.title``, so that during web data extraction the value that
+    ``BookPage.title`` returns gets mapped to ``Book.title``.
+
+
+Use your page object class
+==========================
+
+Now that you have a page object class defined, it is time to use it.
+
+First, install requests_, which is required by ``web_poet.example``.
+
+.. _requests: https://requests.readthedocs.io/en/latest/user/install/
+
+Then copy the following code into
+``tutorial-project/run.py``:
+
+.. literalinclude:: /../tutorial-project/run.py
+   :language: python
+   :lines: 1-5, 7-8
+
+Execute that code:
+
+..  code-block:: bash
+
+    python tutorial-project/run.py
+
+And the ``print(item)`` statement should output the following:
 
 .. code-block:: python
 
-    import requests
+   Book(title='The Exiled')
+
+In this tutorial you use ``web_poet.example.get_item``, which is a simple,
+incomplete implementation of the web-poet standard, built specifically for this
+tutorial, for demonstration purposes. In real projects, use instead an actual
+web-poet framework, like `scrapy-poet`_.
+
+``web_poet.example.get_item`` serves to illustrate the power of web-poet. Once
+you have defined your page object class, a web-poet framework only needs 2
+inputs from you: the URL from which you want to extract data, and the import
+path of the Python modules where you define your page object classes.
```

Comment:
I think we should require 2 inputs:

* URL
* Item type.

Import path to modules should be something which is handled on setup time, not something which is passed to every get_item call. That's global shared state (modules currently imported in the Python process), and we should be explicit about it; it's not good that a call to get_item may implicitly affect other get_item calls.

As for asking to specifiy the item class, it could be error-prone otherwise. For example, the modules may contain page objects which extract different item types from the same pages. Let me elaborate on this.

I think it's quite common that you can't write a specific URL pattern for an item. So, handle_urls could be annotated just with a domain name, without trying to limit it further by path, etc, and that's perfectly fine. In most cases the crawling code knows what kind of content to expect after following the link, so it should be able to use right page objects on right URLs; it may depend no only on the URL, but also on CSS/XPath of the link extracted. 

While being more precise with handle_urls is fine, it may also backfire sometimes, because this logic can get split between crawling (link extraction code) and handle_urls. 

The main purpose for using URL patterns in handle_urls is to distinguish between different page objects which return the same item class, not to infer the item class returned for an URL.